{
	"schema_version": "1.6.0",
	"id": "CVE-2024-12088",
	"modified": "2025-01-14T23:48:28.987691Z",
	"published": "2025-01-14T18:15:25Z",
	"related": [
		"DLA-4015-1",
		"DSA-5843-1"
	],
	"details": "A flaw was found in rsync. When using the `--safe-links` option, rsync fails to properly verify if a symbolic link destination contains another symbolic link within it. This results in a path traversal vulnerability, which may lead to arbitrary file write outside the desired directory.",
	"affected": [
		{
			"package": {
				"ecosystem": "Debian:11",
				"name": "rsync",
				"purl": "pkg:deb/debian/rsync?arch=source"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "3.2.3-4+deb11u2"
						}
					]
				}
			],
			"versions": [
				"3.2.3-4",
				"3.2.3-4+deb11u1"
			],
			"ecosystem_specific": {
				"urgency": "not yet assigned"
			},
			"database_specific": {
				"source": "https://storage.googleapis.com/cve-osv-conversion/osv-output/CVE-2024-12088.json"
			}
		},
		{
			"package": {
				"ecosystem": "Debian:12",
				"name": "rsync",
				"purl": "pkg:deb/debian/rsync?arch=source"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "3.2.7-1+deb12u1"
						}
					]
				}
			],
			"versions": [
				"3.2.7-1"
			],
			"ecosystem_specific": {
				"urgency": "not yet assigned"
			},
			"database_specific": {
				"source": "https://storage.googleapis.com/cve-osv-conversion/osv-output/CVE-2024-12088.json"
			}
		},
		{
			"package": {
				"ecosystem": "Debian:13",
				"name": "rsync",
				"purl": "pkg:deb/debian/rsync?arch=source"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						}
					]
				}
			],
			"versions": [
				"3.2.7-1",
				"3.3.0+ds1-1",
				"3.3.0+ds1-2",
				"3.3.0-1"
			],
			"ecosystem_specific": {
				"urgency": "not yet assigned"
			},
			"database_specific": {
				"source": "https://storage.googleapis.com/cve-osv-conversion/osv-output/CVE-2024-12088.json"
			}
		}
	],
	"references": [
		{
			"type": "REPORT",
			"url": "https://bugzilla.redhat.com/show_bug.cgi?id=2330676"
		},
		{
			"type": "WEB",
			"url": "https://access.redhat.com/security/cve/CVE-2024-12088"
		},
		{
			"type": "WEB",
			"url": "https://kb.cert.org/vuls/id/952657"
		},
		{
			"type": "ADVISORY",
			"url": "https://security-tracker.debian.org/tracker/CVE-2024-12088"
		}
	]
}
